src/bindings/html/dom.js

```diff
@@ -67,51 +69,30 @@ export function translateFragment(view, langs, frag) {
   return translateElements(view, langs, getTranslatables(frag));
 }
 
-function getElementTranslation(view, langs, elem) {
-  const id = elem.getAttribute('data-l10n-id');
+function getElementsTranslation(view, langs, elems) {
+  var keys = [];
 
-  if (!id) {
-    return false;
-  }
-
-  const args = elem.getAttribute('data-l10n-args');
-
-  if (!args) {
-    return view._resolveEntity(langs, id);
-  }
+  elems.forEach(elem => {
+    const id = elem.getAttribute('data-l10n-id');
+    const args = elem.getAttribute('data-l10n-args');
 
-  return view._resolveEntity(
-    langs, id, JSON.parse(
-      args.replace(reHtml, match => htmlEntities[match])));
-}
+    keys.push( args ? [
+      id,
+      JSON.parse(args.replace(reHtml, match => htmlEntities[match]))
+    ] : id);
+  });
 
-export function translateElement(view, langs, elem) {
-  return getElementTranslation(view, langs, elem).then(
-    translation => applyTranslation(view, elem, translation));
+  return view._resolveEntities(langs, keys);
 }
 
 function translateElements(view, langs, elements) {
-  return Promise.all(
-    elements.map(elem => getElementTranslation(view, langs, elem))).then(
-      translations => applyTranslations(view, elements, translations));
-}
-
-function applyTranslation(view, elem, translation) {
```

Comment:
I *really* like that we're making the code simpler by removing `translateElement` and `applyTranslation`!